addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -356,38 +270,142 @@ def _compute_from_employee_id(self):
             elif not holiday.holiday_status_id and not holiday._origin.holiday_status_id:
                 holiday.holiday_status_id = default_holiday_status_id
 
-    @api.depends('holiday_status_id', 'allocation_type', 'number_of_hours_display', 'number_of_days_display')
+    @api.depends('holiday_status_id', 'allocation_type', 'number_of_hours_display', 'number_of_days_display', 'date_to')
     def _compute_from_holiday_status_id(self):
         for allocation in self:
             allocation.number_of_days = allocation.number_of_days_display
             if allocation.type_request_unit == 'hour':
                 allocation.number_of_days = allocation.number_of_hours_display / (allocation.employee_id.sudo().resource_calendar_id.hours_per_day or HOURS_PER_DAY)
+            if allocation.accrual_plan_id.time_off_type_id != allocation.holiday_status_id:
+                allocation.accrual_plan_id = False
+            if allocation.allocation_type == 'accrual' and not allocation.accrual_plan_id and allocation.holiday_status_id:
+                allocation.accrual_plan_id = self.env['hr.leave.accrual.plan'].search([('time_off_type_id', '=', allocation.holiday_status_id.id)], limit=1)
+
+    @api.depends('date_from', 'date_to')
+    def _compute_valid(self):
+        dt = self._context.get('default_date_from') or fields.Date.context_today(self)
+
+        for holiday_type in self:
+            if holiday_type.date_from and holiday_type.date_to:
+                holiday_type.valid = ((dt < holiday_type.date_to) and (dt > holiday_type.date_from))
+            elif holiday_type.date_from and (dt > holiday_type.date_from):
+                holiday_type.valid = False
+            else:
+                holiday_type.valid = True
+
+    def _search_valid(self, operator, value):
+        dt = self._context.get('default_date_from', False)
+
+        if not dt:
```

Comment:
I don't get it.

If no context everything is valid
Else, maybe with a domain that I believe to be wrong.